### ISSUE TITLE
PP occurrence stratum

### ISSUE THREAD

**Issue by sacrevert** [BiologicalRecordsCentre/BSBI-Card-and-PlantPortal-DEPRECATED-#62]
We discussed this in the past, but perhaps it was something that was shelved for simplification, however, it will be important for full functionality of the system.

This is the vertical stratum within which an occurrence/abundance observation is made e.g. a user made record two occurrences for a tree within a plot, one in the ground later (e.g. a seedling) and one in the canopy (i.e. the woodland canopy composition overlapping the plot). This will need to be an additional occurrence attribute with a controlled termlist.

Nested in #46

**Comment by andrewvanbreda** [comment]
Yes, we discussed this. I think it is just missing from the importer as I don't think there is an code to add other than the attribute. I will make a term list and occurrence attribute.
Can you confirm the exact names of the terms you wish to use

**Comment by sacrevert** [comment]
Great, this is an action on me then. I will need to look into the standard codes used in other places and get back to you on this. It may be that we move this up the priorities list, as I cannot load actual data until this is enabled. I'll talk to David about this next week when he is back.

**Comment by sacrevert** [comment]
codes are as follows: c = canopy; s = shrub layer; g = ground layer.
I expect we would ask people to supply them simply as c, s, or g, rather than the full terms.
The column name for mapping could just be 'Stratum'.

These terms would probably need to be concatenated with species names for subsequent downstream operations in WebMAVIS (e.g. Quercus robur (c) -- indicating Oak cover values in the canopy). I know that this is future work, and probably not relevant, but mentioning just in case.

**Comment by andrewvanbreda** [comment]
Stratum and cover have been added to my machine and committed.
Having said that, I have committed that code before reading your comment that people would add them as short characters. I don't think I support that at the moment, I wonder I might look to see if Indicia has built in support for that kind of thing as am sure it is a common issue.
For now, the terms would just be "c" and "s" and "g" which should suffice for testing.
In fact I wonder if the correct way to do this now would be to continue like this but add the full term as a term attribute.
Anyway, I suggest we don't overthink this right now as you say we don't want this to balloon uncontrollably. I'll just keep a note of this here.
As anticipated the fix didn't take long, around 50 minutes including getting the stratum and cover both working and committed.
